parse.go

```diff
@@ -0,0 +1,67 @@
+package guac
+
+import (
+	"errors"
+	"strconv"
+)
+
+func Parse(buf []byte) (*Instruction, error) {
```

Comment:
I think we would prefer if Parse remained in `instruction.go`. That will also make it easier to see what changes you made in this file.

Reply:
Sure thing, no problems! Where do you prefer to have the tests? I used `parse_test.go`, but maybe it's better to have them in `instruction_test.go`? What do you think?

Reply:
Yes, I think they should go in `instruction_test.go`.